src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,265 @@
+        return 'medium';
+    }
+  },
+  fontSize({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xxsmall';
+      case 'small':
+        return 'xsmall';
+      case 'medium':
+      case 'large':
+      default:
+        return 'medium';
+    }
+  },
+  alignSelf({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconPosition }) {
+    switch (size) {
+      case 'xsmall':
+      case 'small':
+        return iconPosition === 'left' ? [0, 0.5, 0, 0] : [0, 0, 0, 0.5];
+      case 'medium':
+      case 'large':
+      default:
+        return iconPosition === 'left' ? [0, 1, 0, 0] : [0, 0, 0, 1];
+    }
+  },
+};
 
-const StyledText = styled(Text)(
+const StyledButton = styled(TouchableHighlight)(
   (props) => `
-  color: ${props.theme.colors.tone[900]};
-  align-self: center;
-  font-size: 18px;
-  letter-spacing: ${props.theme.fonts.letterSpacing.l};
-`,
+    flex-direction: row;
+    align-items: center;
+    align-self: ${styles.alignSelf(props)};
+    background-color: ${styles.backgroundColor(props)};
+    padding: ${styles.padding(props)};
+    border-radius: ${props.theme.spacings.xxsmall};
+    ${styles.border(props) && `border: ${styles.border(props)}`};
+  `,
 );
 
-const Button = ({ onClick, children, ...rest }) => (
-  <StyledButton onPress={onClick} {...rest}>
-    <StyledText {...rest}>{children}</StyledText>
-  </StyledButton>
-);
+const Button = (props) => {
+  const {
+    onClick,
+    children,
+    variant,
+    size,
+    disabled,
+    variantColor,
+    icon,
+    iconPosition,
+    align,
+    testID,
+  } = props;
+  const theme = useContext(ThemeContext);
+  return (
+    <StyledButton
+      onPress={onClick}
+      variantColor={variantColor}
+      disabled={disabled}
+      size={size}
+      variant={variant}
+      activeOpacity={1}
+      text={children}
+      underlayColor={styles.underlayColor({ variantColor, variant, theme })}
+      align={align}
+      testID={testID}
+    >
+      <>
+        {icon && iconPosition === 'left' && (
+          <Icon
+            name={icon}
+            size={styles.iconSize(props)}
+            fill={styles.fontColor({ ...props, theme })}
+          />
+        )}
+        {icon && iconPosition === 'left' && children && (
+          <Space margin={styles.spaceBetween(props)}>
+            <View />
+          </Space>
+        )}
+        <Text
+          color={styles.fontColor(props)}
+          size={styles.fontSize(props)}
+          _weight="bold"
+          align="center"
+          _letterSpacing={styles.letterSpacing(props)}
+        >
+          {size === 'xsmall' ? children && children.toUpperCase() : children}
```

Comment:
Should we do a type check here for children since ```toUpperCase()``` is allowed on strings?

Reply:
In propTypes we have set the type of children to a string. If a person passes anything other than string it won't be a correct usage of API and it will anyways fail.